tools/loader/ExecutorCore.cpp

```diff
@@ -0,0 +1,449 @@
+/**
+ * Copyright (c) Glow Contributors. See CONTRIBUTORS file.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Base/Image.h"
+#include "glow/Converter/TypeAToTypeBFunctionConverter.h"
+#include "glow/Importer/Caffe2ModelLoader.h"
+#include "glow/Importer/ONNXModelLoader.h"
+
+#include "llvm/ADT/StringSwitch.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Format.h"
+#include "llvm/Support/Timer.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include <atomic>
+#include <cfloat>
+#include <fstream>
+#include <future>
+#include <iostream>
+#include <memory>
+#include <mutex>
+#include <queue>
+#include <sstream>
+#include <thread>
+
+#include "ExecutorCore.h"
+#include "ExecutorCoreHelperFunctions.h"
+
+extern llvm::cl::opt<unsigned> traceLevel;
+
+using namespace glow;
+
+class PostProcessingExecutor final {
+public:
+  /// Iterates over registered extensions for processing and Printing results
+  /// and executes them.
+  int processOutputs(const llvm::StringMap<Placeholder *> &PHM,
+                     PlaceholderBindings &bindings,
+                     llvm::ArrayRef<std::string> inputImageBatchFilenames);
+
+  /// Registers Post Processing Output extensions.
+  void registerPostProcessOutputExtensions(
+      const std::vector<
+          std::function<std::unique_ptr<PostProcessOutputDataExtension>()>>
+          &extVector);
+
+private:
+  std::vector<std::unique_ptr<PostProcessOutputDataExtension>> extensions_;
+};
+
+class PreProcessImageExecutor final {
+public:
+  /// Iterates over PreProcessInputDataExtension extensions and executes them
+  /// one by ene.
+  void preProcessInputData(Tensor &inputImageData, size_t startId, size_t endId,
+                           size_t batchSz);
+
+  /// Registers Input Data Preprocessing Extensions.
+  void registerInputDataPreProcessingExtension(
+      const std::vector<
+          std::function<std::unique_ptr<PreProcessInputDataExtension>()>>
+          &extVector);
+
+private:
+  std::vector<std::unique_ptr<PreProcessInputDataExtension>> extensions_;
+};
+
+void PostProcessingExecutor::registerPostProcessOutputExtensions(
+    const std::vector<
+        std::function<std::unique_ptr<PostProcessOutputDataExtension>()>>
+        &extVector) {
+  for (auto &f : extVector) {
+    extensions_.push_back(f());
+  }
+}
+
+/// Iterates over registered extensions for processing and Printing results
+/// and executes them.
```

Comment:
Can you update the doxygen here -- what does it return and what does that return represent?